lib/statsd/instrument/aggregator.rb

```diff
@@ -243,15 +243,21 @@ def datagram_builder(no_prefix:)
       def thread_healthcheck
         @mutex.synchronize do
           unless @flush_thread&.alive?
+            # The main thread is dead, fallback to direct writes
             return false unless Thread.main.alive?
 
+            # The main thread forked, reset the aggregator state
             if @pid != Process.pid
               StatsD.logger.debug { "[#{self.class.name}] Restarting the flush thread after fork" }
+              # Try one last flush to ensure all metrics are sent
+              do_flush
```

Comment:
Hm. I wonder if there's any chance that the flush thread of the parent process gets called after it `fork`s off, thereby resulting in dual writes of the same data point?

...and _if_ that's something that can happen, are there any downsides that we might want to consider? I worry it might cause counter increments to get double-counted, that kind of thing.

However, I haven't really thought too deeply about it, so if that's all impossible or nothing to worry about, then 🎉 this is nice.